sql/src/main/java/io/crate/executor/transport/task/elasticsearch/ESQueryBuilder.java

```diff
@@ -586,7 +586,9 @@ public boolean convert(Function function, Context context) throws IOException {
                 }
 
                 assert left.symbolType() == SymbolType.REFERENCE || left.symbolType() == SymbolType.DYNAMIC_REFERENCE;
-
+                if (DataTypes.isCollectionType(left.valueType()) && DataTypes.isCollectionType(right.valueType())) {
+                    throw new UnsupportedOperationException("Cannot compare two arrays with each other");
+                }
```

Comment:
we should at least add a query for the terms of the array to reduce the number of items to filter